ipu-plugin/pkg/ipuplugin/ipuplugin.go

```diff
@@ -182,33 +216,43 @@ func (s *server) Run() error {
 	return nil
 }
 
-func (s *server) Stop() {
-	s.log.Info("Stopping IPU plugin")
-	if s.mode == types.IpuMode {
-		//Note: Deletes bridge created in EnsureBridgeExists in  Run api.
-		s.bridgeCtlr.DeleteBridges()
-	}
-
-	log.Infof("DeletePhyPortRules, path->%s, 1->%v, 2->%v", s.p4rtClient.GetBin(), AccApfMacList[PHY_PORT0_INTF_INDEX], AccApfMacList[PHY_PORT1_INTF_INDEX])
-	p4rtclient.DeletePhyPortRules(s.p4rtClient, AccApfMacList[PHY_PORT0_INTF_INDEX], AccApfMacList[PHY_PORT1_INTF_INDEX])
+func cleanUpRulesOnExit(p4rtClient types.P4RTClient) error {
+	log.Infof("DeletePhyPortRules, path->%s, 1->%v, 2->%v", p4rtClient.GetBin(), AccApfMacList[PHY_PORT0_INTF_INDEX], AccApfMacList[PHY_PORT1_INTF_INDEX])
+	p4rtclient.DeletePhyPortRules(p4rtClient, AccApfMacList[PHY_PORT0_INTF_INDEX], AccApfMacList[PHY_PORT1_INTF_INDEX])
 
 	vfMacList, err := utils.GetVfMacList()
 	if err != nil {
 		log.Errorf("Stop: Error->%v", err)
+		return fmt.Errorf("Stop: Error->%v", err)
 	}
 	if len(vfMacList) == 0 || (len(vfMacList) == 1 && vfMacList[0] == "") {
 		log.Errorf("No VFs initialized on the host")
 	} else {
-		log.Infof("DeletePeerToPeerP4Rules, path->%s, vfMacList->%v", s.p4rtClient.GetBin(), vfMacList)
-		p4rtclient.DeletePeerToPeerP4Rules(s.p4rtClient, vfMacList)
+		log.Infof("DeletePeerToPeerP4Rules, path->%s, vfMacList->%v", p4rtClient.GetBin(), vfMacList)
+		p4rtclient.DeletePeerToPeerP4Rules(p4rtClient, vfMacList)
 	}
 
-	log.Infof("DeleteLAGP4Rules, path->%s", s.p4rtClient.GetBin())
-	p4rtclient.DeleteLAGP4Rules(s.p4rtClient)
+	log.Infof("DeleteLAGP4Rules, path->%s", p4rtClient.GetBin())
+	p4rtclient.DeleteLAGP4Rules(p4rtClient)
 
-	log.Infof("DeleteRHPrimaryNetworkVportP4Rules, path->%s, 1->%v", s.p4rtClient, AccApfMacList[PHY_PORT1_INTF_INDEX])
-	p4rtclient.DeleteRHPrimaryNetworkVportP4Rules(s.p4rtClient, AccApfMacList[PHY_PORT1_INTF_INDEX])
+	log.Infof("DeleteRHPrimaryNetworkVportP4Rules, path->%s, 1->%v", p4rtClient, AccApfMacList[PHY_PORT1_INTF_INDEX])
+	p4rtclient.DeleteRHPrimaryNetworkVportP4Rules(p4rtClient, AccApfMacList[PHY_PORT1_INTF_INDEX])
+	return nil
+}
 
+func (s *server) Stop() {
+	s.log.Info("Stopping IPU plugin")
+	if s.mode == types.IpuMode {
+		//Note: Deletes bridge created in EnsureBridgeExists in  Run api.
+		s.bridgeCtlr.DeleteBridges()
```

Comment:
How do we ensure this is in sync with DPU operator state?

Reply:
This is older code. Not doing much to it except moved some older functions to a new function to keep it more modular. But good point. We need to harden all the restart cases